platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationManagerImpl.java

```diff
@@ -96,19 +98,20 @@ public ImportMigrationManagerImpl(
           migratables.collect(
               Collectors.toMap(
                   Migratable::getId,
-                  m -> new ImportMigrationContextImpl(report, zip, m),
+                  m -> new ImportMigrationContextImpl(report, zipContents, m),
                   ConfigurationMigrationManager.throwingMerger(),
                   LinkedHashMap::new)); // to preserved ranking order and remove duplicates
       // add a system contexts
-      contexts.put(null, new ImportMigrationContextImpl(report, zip));
-      zip.stream()
-          .filter(ze -> !ze.isDirectory())
-          .map(ze -> new ImportMigrationEntryImpl(this::getContextFor, ze))
+      contexts.put(null, new ImportMigrationContextImpl(report, zipContents));
+      zipContents
+          .entrySet()
+          .stream()
+          .filter(k -> !k.getKey().isDirectory())
+          .map(k -> new ImportMigrationEntryImpl(this::getContextFor, k.getKey()))
```

Comment:
❓ Would it be better to pass in the entry ctor the byte array for the content (or the error if we couldn't read it) as opposed to relying on the entry to get back here to get it later? It was done like that before because we were using the ZipFile and we only wanted to get the input stream on-demand.

Reply:
I meant to ask you about that today actually 😄 . I had a feeling that was why it was done this way originally, but I wanted to check before I just went and changed it.